development-tools/clinic.rst

```diff
@@ -2001,6 +2001,54 @@ The generated glue code looks like this:
 .. versionadded:: 3.13
 
 
+.. _clinic-howto-getter:
+
+How to generate a getter
+------------------------
+
+You can use the ``@getter`` directive to generate an "impl" function
+that defines a `getter <https://docs.python.org/3/c-api/structures.html#c.PyGetSetDef>`_.
+
+This example --- taken from :cpy-file:`Modules/_io/bufferedio.c` ---
+shows the use of ``@getter`` in combination with
+the :ref:`@critical_section <clinic-howto-critical-sections>` directive
+(which achieves thread safety without causing deadlocks between threads)::
+
+    /*[clinic input]
+    @critical_section
+    @getter
+    _io._Buffered.closed
+    [clinic start generated code]*/
+
+The generated glue code looks like this:
+
+.. code-block:: c
+
+    static PyObject *
+    _io__Buffered_closed_get(buffered *self, void *context)
+    {
+        PyObject *return_value = NULL;
+
+        Py_BEGIN_CRITICAL_SECTION(self);
+        return_value = _io__Buffered_closed_get_impl(self);
```

Comment:
There doesn't seem to be an `_io._Buffered` class at runtime AFAICS, and in the Python REPL example below you use `_io._BufferedIOBase`. Did you maybe mean to write this?

```suggestion
    _io._BufferedIOBase.closed
    [clinic start generated code]*/

The generated glue code looks like this:

.. code-block:: c

    static PyObject *
    _io__BufferedIOBase_closed_get(buffered *self, void *context)
    {
        PyObject *return_value = NULL;

        Py_BEGIN_CRITICAL_SECTION(self);
        return_value = _io__BufferedIOBase_closed_get_impl(self);
```

Reply:
Well, that's because..

https://github.com/python/cpython/blob/07ebd46f9e55ed2f18c5ea2a79ec5054bc26b915/Modules/_io/bufferedio.c#L2529

https://github.com/python/cpython/blob/07ebd46f9e55ed2f18c5ea2a79ec5054bc26b915/Modules/_io/bufferedio.c#L2589

https://github.com/python/cpython/blob/07ebd46f9e55ed2f18c5ea2a79ec5054bc26b915/Modules/_io/bufferedio.c#L2707

Reply:
I see, thanks! Feel free to ignore me then (though if we can think of an example that _does_ have a one-to-one correlation with a Python object at runtime, that would be perfect! :)